examples/examples/tower-load.rs

```diff
@@ -1,410 +1,379 @@
+            HandleError::Unknown => f.pad("unknown internal error"),
+        }
+    }
+}
+
+impl std::error::Error for HandleError {}
+
+impl fmt::Display for WrongMethod {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "unsupported method: please use one of {:?}", self.0)
+    }
+}
+
+impl std::error::Error for WrongMethod {}
+
+fn gen_uri(authority: &str) -> (usize, String) {
+    static ALPHABET: &str = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    let mut rng = rand::thread_rng();
+    let idx = rng.gen_range(0, ALPHABET.len() + 1);
+    let len = rng.gen_range(0, 26);
+    let letter = ALPHABET.get(idx..=idx).unwrap_or("");
+    (len, format!("http://{}/{}", authority, letter))
+}
+
+// .instrument(tracing::info_span!(target: "gen", "load_gen", remote.addr=%addr));
+#[tracing::instrument]
+async fn load_gen(addr: &SocketAddr) -> Result<(), Err> {
+    let svc = ServiceBuilder::new()
+        .buffer(5)
+        .layer(request_span::layer(req_span))
+        .timeout(Duration::from_millis(200))
+        .service(Client::new());
+
+    while let Some(_) = time::interval(Duration::from_millis(50)).next().await {
+        let authority = format!("{}", addr);
+        let mut svc = svc.clone();
+        svc.ready().await?;
+        let f = async move {
+            let sleep = rand::thread_rng().gen_range(0, 25);
+            time::delay_for(Duration::from_millis(sleep)).await;
+
+            let (len, uri) = gen_uri(&authority);
+            let req = Request::get(&uri[..])
+                .header("Content-Length", len)
+                .body(Body::empty())
+                .unwrap();
+
+            let rsp = match svc.call(req).await {
+                Err(e) => {
+                    error!(target: "gen", message = "request error!", error = %e);
+                    return Err(e);
+                }
+                Ok(rsp) => rsp,
+            };
+
+            let status = rsp.status();
+            if status != StatusCode::OK {
+                error!(target: "gen", message = "error received from server!", ?status);
+            }
+
+            let body = match hyper::body::to_bytes(rsp).await {
+                Err(e) => {
+                    error!(target: "gen", message = "body error!", error = ?e);
+                    return Err(e.into());
+                }
+                Ok(body) => body,
+            };
+            let body = String::from_utf8(body.to_vec())?;
+            info!(target: "gen", message = "response complete.", rsp.body = %body);
+            Ok(())
+        }
+        .instrument(span!(target: "gen", Level::INFO, "load_gen", remote.addr=%addr));
+        tokio::spawn(f);
+    }
+
+    Ok(())
+}
+
+fn req_span<A>(req: &Request<A>) -> Span {
+    let span = tracing::span!(
+        target: "gen",
+        Level::INFO,
+        "request",
+        req.method = ?req.method(),
+        req.path = ?req.uri().path(),
+        headers = ?req.headers()
```

Comment:
it looks like you added the headers here? the old example didn't put them on the span for brevity reasons, and just recorded them in the first event when receiving a request...since that wasn't removed, we now get logs like this:
```
Feb 10 14:43:47.463 DEBUG request{req.method=GET req.path="/b" headers={"content-length": "12", "host": "[::1]:3000"}}: tower_load: received request. req.headers={"content-length": "12", "host": "[::1]:3000"} req.version=HTTP/1.1
Feb 10 14:43:47.463 DEBUG request{req.method=GET req.path="/s" headers={"content-length": "2", "host": "[::1]:3000"}}: tower_load: received request. req.headers={"content-length": "2", "host": "[::1]:3000"} req.version=HTTP/1.1
Feb 10 14:43:47.464 DEBUG request{req.method=GET req.path="/s" headers={"content-length": "6", "host": "[::1]:3000"}}: tower_load: received request. req.headers={"content-length": "6", "host": "[::1]:3000"} req.version=HTTP/1.1
```
which includes the headers _twice_ and is kinda gross...

i have a slight preference for doing it the way the old implementation did it, but if you think it's important to record the headers on the span, we should probably remove them from the "received request" event. 

also, the other fields are namespaced in `req.$WHATEVER`, can we do that for the headers as well (or nix that convention, i don't care)?

Reply:
> it looks like you added the headers here? the old example didn't put them on the span for brevity reasons, and just recorded them in the first event when receiving a request...since that wasn't removed, we now get logs like this: [example redacted]

Yeah, that's an oversight. I've removed the headers from the span.